MaxText/inference_microbenchmark_sweep.py

```diff
@@ -0,0 +1,94 @@
+"""
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+"""Sweep across inference microbenchmarks."""
+
+import os
+import sys
+import json
+import jsonlines
+import inference_microbenchmark
+import pyconfig
+from jax._src.lib import xla_extension
+
+
+def main():
+  pyconfig.initialize(sys.argv)
+  config = pyconfig.config
+
+  inference_microbenchmark_sweep_key_value_axis_order_product_id_list = [
+    item for item in config.inference_microbenchmark_sweep_key_value_axis_order_product_id_list.split(':')
+  ]
+  inference_microbenchmark_sweep_ar_key_axis_order_list = [
+    item for item in config.inference_microbenchmark_sweep_ar_key_axis_order_list.split(':')
+  ]
+  inference_microbenchmark_sweep_ar_value_axis_order_list = [
+    item for item in config.inference_microbenchmark_sweep_ar_value_axis_order_list.split(':')
+  ]
+
+  results = []
+  for (
+    key_value_axis_order_product_id,
+    ar_key_axis_order,
+    ar_value_axis_order,
+  ) in zip(
+    inference_microbenchmark_sweep_key_value_axis_order_product_id_list,
+    inference_microbenchmark_sweep_ar_key_axis_order_list,
+    inference_microbenchmark_sweep_ar_value_axis_order_list,
+  ):
+    print(f"key_value_axis_order_product_id {key_value_axis_order_product_id}")
+    print(f"ar_key_axis_order {ar_key_axis_order}")
+    print(f"ar_value_axis_order {ar_value_axis_order}")
+
+    # Manually update the config
+    # Don't set key_value_axis_order_product_id; otherwise it will recompute ar_key_axis_order and ar_value_axis_order
+    pyconfig._config.keys['ar_key_axis_order'] = ar_key_axis_order
+    pyconfig._config.keys['ar_value_axis_order'] = ar_value_axis_order
+    name = f"{key_value_axis_order_product_id}-{ar_key_axis_order}-{ar_value_axis_order}"
+    tensorboard_dir = os.path.join(config.base_output_directory, name, "tensorboard", "")
+    pyconfig._config.keys['tensorboard_dir'] = tensorboard_dir
+
+    print(f"ar_key_axis_order {config.ar_key_axis_order}")
+    print(f"ar_value_axis_order {config.ar_value_axis_order}")
+    dimensions_json = {}
```

Comment:
If this dimensions_json is for inference only, we can move it to https://github.com/google/maxtext/blob/main/MaxText/inference_utils.py

And have an api to let inference_microbenchmark.py write into it.